src/modinfo.c

```diff
@@ -1314,8 +1316,12 @@ sr_module_oper_data_load(struct sr_mod_info_mod_s *mod, sr_conn_ctx_t *conn, uin
             last_sid = 1;
         }
         if (!last_sid || !mod_oper_data) {
-            /* load push oper data for the session */
-            if ((err_info = sr_module_file_data_append(mod->ly_mod, mod->ds_handle, SR_DS_OPERATIONAL,
+            if ((oper_push_dup[i].sid == sid) && sess->oper_push_cache) {
+                /* load push oper data from the session cache (consume it) */
+                mod_data = sr_module_data_unlink(&sess->oper_push_cache, mod->ly_mod, 1);
+            }
+            /* load push oper data for the session from the datastore plugin if no data loaded above */
```

Comment:
I think the situation with no push oper data of this session would not work correctly. Could perhaps be fixed by moving `oper_push_cache` in the session to `oper_push_mods`. These can be separated by module unlike `running` data because they are never used as one large data tree but always only erged or something similar to the `operational` DS data.